pkgs/applications/science/logic/rocq/default.nix

```diff
@@ -0,0 +1,152 @@
+        (defun nixpkgs--around-scomint-make (orig &rest r)
+          (if nixpkgs--is-nixpkgs-rocq-buffer
+              (progn
+                (advice-add 'get-buffer-create :filter-return #'nixpkgs--rocq-inherit)
+                (apply orig r)
+                (advice-remove 'get-buffer-create #'nixpkgs--rocq-inherit))
+            (apply orig r)))
+        (advice-add 'scomint-make :around #'nixpkgs--around-scomint-make)
+        ; When we have no more rocq buffers, tear down the buffer handling
+        (defun nixpkgs--decrement-rocq-buffer-count ()
+          (setq nixpkgs--rocq-buffer-count (1- nixpkgs--rocq-buffer-count))
+          (when (eq nixpkgs--rocq-buffer-count 0)
+            (advice-remove 'scomint-make #'nixpkgs--around-scomint-make)
+            (fmakunbound 'nixpkgs--around-scomint-make)
+            (fmakunbound 'nixpkgs--rocq-inherit)
+            (fmakunbound 'nixpkgs--decrement-rocq-buffer-count))))
+      (setq nixpkgs--rocq-buffer-count (1+ nixpkgs--rocq-buffer-count))
+      (add-hook 'kill-buffer-hook 'nixpkgs--decrement-rocq-buffer-count nil t)
+      (setq nixpkgs--is-nixpkgs-rocq-buffer t)
+      (inherit-local 'nixpkgs--is-nixpkgs-rocq-buffer)
+    '';
+  };
+
+  nativeBuildInputs = [ pkg-config ]
+    ++ ocamlNativeBuildInputs;
+  buildInputs = [ ncurses ];
+
+  propagatedBuildInputs = ocamlPropagatedBuildInputs;
+
+  postPatch = ''
+    UNAME=$(type -tp uname)
+    RM=$(type -tp rm)
+    substituteInPlace tools/beautify-archive --replace "/bin/rm" "$RM"
+    ${csdpPatch}
+  '';
+
+  setupHook = writeText "setupHook.sh" ''
+    addCoqPath () {
+      if test -d "''$1/lib/coq/${rocq-version}/user-contrib"; then
+        export COQPATH="''${COQPATH-}''${COQPATH:+:}''$1/lib/coq/${rocq-version}/user-contrib/"
+      fi
+    }
+
+    addEnvHooks "$targetOffset" addCoqPath
+  '';
+
+  preConfigure = ''
+    patchShebangs dev/tools/
+  '';
+
+  prefixKey = "-prefix ";
+
+  enableParallelBuilding = true;
+
+  createFindlibDestdir = true;
+
+  buildPhase = ''
+    runHook preBuild
+    make dunestrap
+    dune build -p rocq-runtime,rocq-core -j $NIX_BUILD_CORES
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    dune install --prefix $out rocq-runtime rocq-core
+    ln -s $out/lib/rocq-runtime $OCAMLFIND_DESTDIR/rocq-runtime
+    ln -s $out/lib/rocq-core $OCAMLFIND_DESTDIR/rocq-core
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "The Rocq Prover";
+    longDescription = ''
+      The Rocq Prover is an interactive theorem prover, or proof assistant. It provides
+      a formal language to write mathematical definitions, executable
+      algorithms and theorems together with an environment for
+      semi-interactive development of machine-checked proofs.
+    '';
+    homepage = "https://coq.inria.fr";
+    license = licenses.lgpl21;
+    branch = rocq-version;
+    maintainers = with maintainers; [ roconnor thoughtpolice vbgl Zimmi48 ];
```

Comment:
Please let these people know. I agree to be listed there. You (Pierre) may want to add yourself to the list.

Reply:
@roconnor @thoughtpolice @Zimmi48 is it fine to keep you here? (maintainers list of rocq-core package (renaming of Coq))

Reply:
I'm happy to stay in.  However if you folks prefer to remove me since I haven't contributed in years, I would understand.

Reply:
Happy to be listed as well.